cpp/src/community/detail/mis_impl.cuh

```diff
@@ -102,10 +92,7 @@ rmm::device_uvector<vertex_t> compute_mis(
   out_degrees.resize(0, handle.get_stream());
   out_degrees.shrink_to_fit(handle.get_stream());
 
-  thrust::default_random_engine g;
-  size_t seed = 0;
-  if constexpr (multi_gpu) { seed = handle.get_comms().get_rank(); }
-  g.seed(seed);
+  raft::random::RngState rng_state(multi_gpu ? handle.get_comms().get_rank() : 0);
```

Comment:
This should be a parameter to the `compute_mis` function rather than being constructed here.

Reply:
Yes, and if run Leiden multiple times with the same `rng_state`, we will get different results in each run and this will be helpful in implementing ECG as well.

Reply:
Resolved.